JsonLd/ContextBuilder.php

```diff
@@ -119,7 +126,11 @@ public function getContext(ResourceInterface $resource = null)
             }
         }
 
-        return $context;
+        // Dispatch event to allow context customization
+        $event = new Event\ContextBuilderEvent($context);
```

Comment:
IMO, here's the content of that `getContent` method:
```php
$context = $this->getBaseContext();
$event = new Event\ContextBuilderEvent($context, $resource);
$this->eventDispatcher->dispatch(Event\Events::CONTEXT_BUILDER, $event);
return $event->getContext();
```

Then, you just have to move what's in the `if ($resource) {` in a listener.